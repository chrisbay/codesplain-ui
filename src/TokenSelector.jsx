import React from 'react';
import { List, ListItem } from 'material-ui/List';
import Subheader from 'material-ui/Subheader';
import Checkbox from 'material-ui/Checkbox';

class TokenSelector extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      selectedTokens: [],
    };
  }
  render() {
    return (
      <List>
        <Subheader>Token types to highlight</Subheader>
        <ListItem
          leftCheckbox={<Checkbox />}
          primaryText="Functions"
        />
        <ListItem
          leftCheckbox={<Checkbox />}
          primaryText="Variables"
        />
        <ListItem
          leftCheckbox={<Checkbox />}
          primaryText="For loop"
        />
      </List>
    );
  }

}

export default TokenSelector;
