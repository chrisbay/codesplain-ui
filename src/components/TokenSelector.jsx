import React, { PropTypes } from 'react';
import { List, ListItem } from 'material-ui/List';
import Subheader from 'material-ui/Subheader';
import Checkbox from 'material-ui/Checkbox';
import '../styles/codesplain.css';

class TokenSelector extends React.Component {
  constructor(props) {
    super(props);
    this.itemState = {}
    this.makeListItems = this.makeListItems.bind(this);
  }

  makeListItems() {
    return this.props.tokenTypes.map((tokenType, index) => {
      const tokenText = tokenType.text
      if (this.itemState[tokenText] === undefined)
        this.itemState[tokenText] = false;
      return (
        <ListItem
          className="style-listheight"
          key={`${tokenText}-index`}
          leftCheckbox={
            <Checkbox onCheck={
                (ev, checked) => this.itemState[tokenText] = checked
              }
              className="styles-checkbox"
            />
          }
          primaryText={tokenText}
        />
      );
    });
  }

  render() {
    return (
      <List  onChange={() => this.props.onChange(this.itemState)}>
        <Subheader>Token types to highlight</Subheader>
        { this.makeListItems() }
      </List>
    );
  }
}

TokenSelector.propTypes = {
  tokenTypes: PropTypes.arrayOf(PropTypes.shape({
    text: PropTypes.string,
  })).isRequired,
  onChange: PropTypes.func.isRequired,
};
export default TokenSelector;
