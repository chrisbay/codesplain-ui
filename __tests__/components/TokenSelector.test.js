import React from 'react'
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

import TokenSelector from '../../src/components/TokenSelector';

jest.mock('material-ui/Checkbox')

const mockFilters = {
  'for_stmt': {
    prettyTokenName: 'For Loops',
    count: 1,
    selected: false
  },
  'atom': {
    prettyTokenName: 'Atoms',
    count: 21,
    selected: true
  }
}

describe('<TokenSelector />', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(
      <MuiThemeProvider>
        <TokenSelector
          tokens={mockFilters}
          onChange={jest.fn()}
        />
      </MuiThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  })
})
