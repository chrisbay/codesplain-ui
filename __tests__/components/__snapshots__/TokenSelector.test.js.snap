exports[`<TokenSelector /> matches snapshot 1`] = `
<div
  onChange={[Function]}
  style={
    Object {
      "muiPrepared": true,
      "padding": "0px 0px 8px 0px",
    }
  }>
  <div
    style={
      Object {
        "boxSizing": "border-box",
        "color": "rgba(0, 0, 0, 0.54)",
        "fontSize": 14,
        "fontWeight": 500,
        "lineHeight": "48px",
        "muiPrepared": true,
        "paddingLeft": 16,
        "width": "100%",
      }
    }>
    Token types to highlight
  </div>
  <div>
    <label
      className="style-listheight"
      onTouchEnd={[Function]}
      open={null}
      style={
        Object {
          "backgroundColor": null,
          "color": "rgba(0, 0, 0, 0.87)",
          "cursor": "pointer",
          "display": "block",
          "fontSize": 16,
          "lineHeight": "16px",
          "marginLeft": 0,
          "muiPrepared": true,
          "paddingBottom": 16,
          "paddingLeft": 72,
          "paddingRight": 16,
          "paddingTop": 16,
          "position": "relative",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        }
      }>
      <div
        style={
          Object {
            "muiPrepared": true,
          }
        }>
        for-each
      </div>
    </label>
  </div>
  <div>
    <label
      className="style-listheight"
      onTouchEnd={[Function]}
      open={null}
      style={
        Object {
          "backgroundColor": null,
          "color": "rgba(0, 0, 0, 0.87)",
          "cursor": "pointer",
          "display": "block",
          "fontSize": 16,
          "lineHeight": "16px",
          "marginLeft": 0,
          "muiPrepared": true,
          "paddingBottom": 16,
          "paddingLeft": 72,
          "paddingRight": 16,
          "paddingTop": 16,
          "position": "relative",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        }
      }>
      <div
        style={
          Object {
            "muiPrepared": true,
          }
        }>
        for-in
      </div>
    </label>
  </div>
  <div>
    <label
      className="style-listheight"
      onTouchEnd={[Function]}
      open={null}
      style={
        Object {
          "backgroundColor": null,
          "color": "rgba(0, 0, 0, 0.87)",
          "cursor": "pointer",
          "display": "block",
          "fontSize": 16,
          "lineHeight": "16px",
          "marginLeft": 0,
          "muiPrepared": true,
          "paddingBottom": 16,
          "paddingLeft": 72,
          "paddingRight": 16,
          "paddingTop": 16,
          "position": "relative",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        }
      }>
      <div
        style={
          Object {
            "muiPrepared": true,
          }
        }>
        for-of
      </div>
    </label>
  </div>
</div>
`;
